src/keyvalue-document.ts

```diff
@@ -145,39 +145,52 @@ export class KeyvalueDocumentFormatter implements DocumentFormattingEditProvider
         const tokens = kvDoc.getAllTokens();
         
         let indentation = 0;
-        const edits: TextEdit[] = [];
 
-        for(let i = 0; i < tokens.length; i++) {
+        const startPos = document.positionAt(0);
+        
+        // We just delete and reconstruct the entire file. Much easier.
+        const edits: TextEdit[] = [ 
+            new TextEdit( new Range(startPos, document.positionAt(document.getText().length)), "" )
+        ];
+
+        const bracketOnNewline = this.doPutBracesOnNewline();
 
+        let text = "";
+
+        for(let i = 0; i < tokens.length; i++) {
             const token = tokens[i];
```

Comment:
```suggestion
        for(const token of tokens) {
```

Reply:
I'll keep the for loop in case i want to access the next or previous item at some point